src/hooks/sdk/useMediaStreamMetrics.tsx

```diff
@@ -41,43 +49,29 @@ export function useMediaStreamMetrics(): MediaStreamMetrics {
         const {
           audioPacketLossPercent,
           audioPacketsReceivedFractionLoss,
+          audioSpeakerDelayMs,
+          audioUpstreamRoundTripTimeMs,
+          audioUpstreamJitterMs,
+          audioDownstreamJitterMs,
+          currentRoundTripTimeMs,
           availableOutgoingBitrate,
           availableIncomingBitrate,
         } = clientMetricReport.getObservableMetrics();
-        let videoStreamMetrics = {};
-        let availableOutgoingBandwidth = 0;
-        let availableIncomingBandwidth = 0;
-        let audioPacketsSentFractionLossPercent = 0;
-        let audioPacketsReceivedFractionLossPercent = 0;
-
-        if (isValidMetric(audioPacketLossPercent)) {
-          audioPacketsSentFractionLossPercent = Math.trunc(
-            audioPacketLossPercent
-          );
-        }
-
-        if (isValidMetric(audioPacketsReceivedFractionLoss)) {
-          audioPacketsReceivedFractionLossPercent = Math.trunc(
-            audioPacketsReceivedFractionLoss
-          );
-        }
-
-        videoStreamMetrics = clientMetricReport.getObservableVideoMetrics();
-
-        if (isValidMetric(availableOutgoingBitrate)) {
-          availableOutgoingBandwidth = availableOutgoingBitrate / 1000;
-        }
-
-        if (isValidMetric(availableIncomingBitrate)) {
-          availableIncomingBandwidth = availableIncomingBitrate / 1000;
-        }
 
+        // Return 0 if the metric value is NaN, otherwise return its integer part.
         setMediaStreamMetrics({
-          audioPacketsSentFractionLossPercent,
-          audioPacketsReceivedFractionLossPercent,
-          availableOutgoingBandwidth,
-          availableIncomingBandwidth,
-          videoStreamMetrics,
+          audioPacketsSentFractionLossPercent: audioPacketLossPercent | 0,
```

Comment:
No `Math.trunc` anymore here? Earlier see that we use `Math.trunc`.

Reply:
I knew this could be  confusing so I left comment above to explain this:
>// Return 0 if the metric value is NaN, otherwise return its integer part.

The bitwise OR operator truncates the value.
for example:

```js
NaN | 0 = 0
undefined | 0 = 0
1 | 0 = 1
1.1 | 0 = 1
```

Reply:
So, this is hard to understand and readers or contributors may get confused, to keep this simple can we simply use the `Math.trunc`?

Reply:
Already left comment for this, don't think it's necessary to add extra `Math.trunc`. I would like to change this if you have strong opinion, but personally I prefer not. Don't think this redundancy benefits too much.

```js
setMediaStreamMetrics({
  audioPacketsSentFractionLossPercent:
    Math.trunc(audioPacketLossPercent) | 0,
  audioPacketsReceivedFractionLossPercent:
    Math.trunc(audioPacketsReceivedFractionLoss) | 0,
  audioSpeakerDelayMs: Math.trunc(audioSpeakerDelayMs) | 0,
  audioUpstreamRoundTripTimeMs:
    Math.trunc(audioUpstreamRoundTripTimeMs) | 0,
  audioUpstreamJitterMs: Math.trunc(audioUpstreamJitterMs) | 0,
  audioDownstreamJitterMs: Math.trunc(audioDownstreamJitterMs) | 0,
  currentRoundTripTimeMs: Math.trunc(currentRoundTripTimeMs) | 0,
  availableOutgoingBandwidth:
    Math.trunc(availableOutgoingBitrate / 1000) | 0,
  availableIncomingBandwidth:
    Math.trunc(availableIncomingBitrate / 1000) | 0,
  rtcStatsReport: clientMetricReport.getRTCStatsReport(),
  videoStreamMetrics: clientMetricReport.getObservableVideoMetrics(),
});
```

```js
setMediaStreamMetrics({
  audioPacketsSentFractionLossPercent: audioPacketLossPercent | 0,
  audioPacketsReceivedFractionLossPercent:
    audioPacketsReceivedFractionLoss | 0,
  audioSpeakerDelayMs: audioSpeakerDelayMs | 0,
  audioUpstreamRoundTripTimeMs: audioUpstreamRoundTripTimeMs | 0,
  audioUpstreamJitterMs: audioUpstreamJitterMs | 0,
  audioDownstreamJitterMs: audioDownstreamJitterMs | 0,
  currentRoundTripTimeMs: currentRoundTripTimeMs | 0,
  availableOutgoingBandwidth: (availableOutgoingBitrate / 1000) | 0,
  availableIncomingBandwidth: (availableIncomingBitrate / 1000) | 0,
  rtcStatsReport: clientMetricReport.getRTCStatsReport(),
  videoStreamMetrics: clientMetricReport.getObservableVideoMetrics(),
});
```

Reply:
Do you have to trunc for units in `Ms` or truncing only applies to the percent metrics?

Reply:
>Do you have to trunc for units in Ms or truncing only applies to the percent metrics?

In the previous version, all metrics are truncated. 

We did something like this:

```js
const { audioPacketLossPercent } = clientMetricReport.getObservableMetrics();

// 1. Set the default value 0
let audioPacketsSentFractionLossPercent = 0;

// 2. Validate metric value
if (isValidMetric(audioPacketLossPercent)) {
  audioPacketsSentFractionLossPercent = Math.trunc(
    audioPacketLossPercent
  );
}

// 3. Return
setMediaStreamMetrics({
  audioPacketsSentFractionLossPercent
)}
```
It's quite redundant if there are more and more metrics.

The main goal here to use `|` is to merge the previous step 1 and step 2. To simplify the logic.